### PR TITLE
LPS-148702 Fix ResourcePermission deletion for Announcements and Roles - LPS-145396 Subtask

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
When we delete some types of objects in Liferay, we are not deleting their ResourcePermission records, leaving some database wrong references.

For more information see parent LPS-145396

This subtask addresses the issue for User and System Mngt. team code

/cc @tinatian

### CODE REVIEW

File: portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java [hunk 1]
```diff
@@ -461,14 +461,9 @@ public Role deleteRole(Role role) throws PortalException {
...
 		}
 
-		String className = role.getClassName();
-		long classNameId = role.getClassNameId();
-
-		if ((classNameId <= 0) || className.equals(Role.class.getName())) {
-			_resourceLocalService.deleteResource(
-				role.getCompanyId(), Role.class.getName(),
-				ResourceConstants.SCOPE_INDIVIDUAL, role.getRoleId());
-		}
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:471]: @jorgediaz-lr do you know what this `if` check was supposed to accomplish? I'm not sure myself, and this change seems safe enough, but I just wanted to double-check :)

Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:471]: @jorgediaz-lr do you know what this `if` check was supposed to accomplish? I'm not sure myself, and this change seems safe enough, but I just wanted to double-check :)

**Review by drewbrokke — APPROVED** [review]

**Comment by drewbrokke** [comment]
ci:forward

File: portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java (hunk 1 above)
Comment (jorgediaz-lr) [reply portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:471]: Hi @drewbrokke 

I did some analysis and I found that the code comes from very old versions of Liferay:
 - The `className.equals(Role.class.getName())` check was added in the LPS-2401, see https://github.com/liferay/liferay-portal/commit/b7b9e50ef70dc07dc37e722ee136ccb8cfaea743#diff-0bb5a1b2e49bed884487582f880c6cb5fb54b44acc07b1d7d3e6558d5fe86b48R215
 - The original IF was added in a commit added in 2006 without linked JIRA ticket, see https://github.com/liferay/liferay-portal/commit/e14a11a1059442facebdbde6bd311d1154924aa8#diff-3cff40f492b9f96833e1ba4cf57029711f8804eca534c2aec3df98082fa26a66R118

But I didn't found the real objective of this code.

**Pull request closed without merging**